Fix deprecated code snippets and removed pre-6.3 code snippets (#233)
smartdevicelink/sdl_app_library_guides: docs/Speech and Audio/Getting Microphone Audio/index.md

!!!

## Starting Audio Capture

> ### Getting the Supported Capabilities

Reviewer: There should still be some sort of intro between the ## and ###

Author: Added

You must use a sampling rate, bit rate, and audio type supported by the head unit. Once you have successfully connecting to the module, you can access these properties on the @![iOS]`SDLManager.systemCapabilityManager`!@@![android, javaSE, javaEE]`sdlManager.getSystemCapabilityManager`!@ instance.

@![iOS]
##### Objective-C
```objc
NSArray<SDLAudioPassThruCapabilities *> *audioPassThruCapabilities = self.sdlManager.systemCapabilityManager.audioPassThruCapabilities;
```

##### Swift
```swift
let audioPassThruCapabilities = sdlManager.systemCapabilityManager.audioPassThruCapabilities
```
!@

@![android, javaSE, javaEE]
```java
sdlManager.getSystemCapabilityManager().getCapability(SystemCapabilityType.AUDIO_PASSTHROUGH, new OnSystemCapabilityListener() {
	@Override
	public void onCapabilityRetrieved(Object capability) {

> 		List<AudioPassThruCapabilities> audioPassThruCapabilities = (List<AudioPassThruCapabilities>) capability;

Reviewer: Spacing is incorrect

Author: Should be fixed

!@

@![javascript]
```js
let audioPassThruCapabilities = sdlManager.getSystemCapabilityManager()._audioPassThruCapabilities;
```
!@

The module may return one or multiple supported audio pass thru capabilities. Each capability will have the following properties:

| Audio Pass Thru Capability | Parameter Name  |  Description |
| ------------- | ------------- | ------------- |
| Sampling Rate | samplingRate | The sampling rate |
| Bits Per Sample | bitsPerSample | The sample depth in bits |
| Audio Type | audioType | The audio type |

### Sending the Audio Capture Request
To initiate audio capture, first construct a @![iOS]`SDLPerformAudioPassThru`!@@![android,javaSE,javaEE,javascript]`PerformAudioPassThru`!@ request. 

@![iOS]
##### Objective-C
```objc
SDLPerformAudioPassThru *audioPassThru = [[SDLPerformAudioPassThru alloc] initWithInitialPrompt:@"<#A speech prompt when the dialog appears#>" audioPassThruDisplayText1:@"<#Ask me \"What's the weather?\"#>" audioPassThruDisplayText2:@"<#or \"What is 1 + 2?\"#>" samplingRate:SDLSamplingRate16KHZ bitsPerSample:SDLBitsPerSample16Bit audioType:SDLAudioTypePCM maxDuration:<#Time in milliseconds to keep the dialog open#> muteAudio:YES];

> 

Author: Added back

[self.sdlManager sendRequest:audioPassThru];
```

##### Swift
```swift
let audioPassThru = SDLPerformAudioPassThru(initialPrompt: "<#A speech prompt when the dialog appears#>", audioPassThruDisplayText1: "<#Ask me \"What's the weather?\"#>", audioPassThruDisplayText2: "<#or \"What is 1 + 2?\"#>", samplingRate: .rate16KHZ, bitsPerSample: .sample16Bit, audioType: .PCM, maxDuration: <#Time in milliseconds to keep the dialog open#>, muteAudio: true)

> 

Author: Added back.

> sdlManager.send(audioPassThru)

Author: Fixed. Added

!!!

## Starting Audio Capture

> Before you start an audio capture session you need to find out what audio pass pass thru capabilities the module supports. You can then use that information to start an audio pass thru session.

Author: Fixed

### Getting the Supported Capabilities
You must use a sampling rate, bit rate, and audio type supported by the module. Once you have successfully connected to the module, you can access these properties on the @![iOS]`SDLManager.systemCapabilityManager`!@@![android,javaSE,javaEE,javascript]`sdlManager.getSystemCapabilityManager`!@ instance.

@![iOS]
##### Objective-C
```objc
NSArray<SDLAudioPassThruCapabilities *> *audioPassThruCapabilities = self.sdlManager.systemCapabilityManager.audioPassThruCapabilities;
```

##### Swift
```swift
let audioPassThruCapabilities = sdlManager.systemCapabilityManager.audioPassThruCapabilities
```
!@

@![android, javaSE, javaEE]
```java
sdlManager.getSystemCapabilityManager().getCapability(SystemCapabilityType.AUDIO_PASSTHROUGH, new OnSystemCapabilityListener() {
    @Override
    public void onCapabilityRetrieved(Object capability) {
        List<AudioPassThruCapabilities> audioPassThruCapabilities = (List<AudioPassThruCapabilities>) capability;
    }

    @Override
    public void onError(String info) {
        // Handle Error
    }
}, false);
```
!@

@![javascript]
```js

> // This is technically a private property and will be made public by removing the `_` in a future release

Reviewer 2: ```suggestion
// This is technically a private property and a `getAudioPassThruCapabilities` method will be added to retrieve it in a future release.
```

performAPT.setAudioType(AudioType.PCM);
performAPT.setMuteAudio(false);

sdlManager.addOnRPCNotificationListener(FunctionID.ON_AUDIO_PASS_THRU, new OnRPCNotificationListener() {
    @Override
    public void onNotified(RPCNotification notification) {
        OnAudioPassThru onAudioPassThru = (OnAudioPassThru) notification;
        // Process audio data
    }
});

> sdlManager.sendRPC(performAPT);

Reviewer: This also needs to handle the response like the iOS code

Author: Should be fixed now.

#### Format of Audio Data
The format of audio data is described as follows:

> 

Reviewer: ```suggestion
```
